Remove `lifetime` from the `Event` (#2981)
rust-windowing/winit: CHANGELOG.md

# Unreleased

> - **Breaking:** Remove `PartialEq` implementation from the event.
> - **Breaking:** Remove `lifetime` from the event.

Reviewer: ```suggestion
- **Breaking:** Remove `PartialEq` implementation from `Event` and `WindowEvent`.
- **Breaking:** Remove lifetime parameter from `Event` and `WindowEvent`.
```
